diff --git a/hosts/android/src/terminal.rs b/hosts/android/src/terminal.rs
--- a/hosts/android/src/terminal.rs
+++ b/hosts/android/src/terminal.rs
@@ -87,8 +87,10 @@
     fn drop(&mut self) {
         #[cfg(target_os = "android")]
         unsafe {
-            libc::kill(self.pid, libc::SIGHUP);
-            libc::kill(self.pid, libc::SIGKILL);
+            if self.exitCode.is_none() {
+                libc::kill(self.pid, libc::SIGHUP);
+                libc::kill(self.pid, libc::SIGKILL);
+            }
             let fd = match self.writer.lock() {
                 Ok(mut writerFd) => {
                     let value = *writerFd;
